File: src/main/java/de/steklopod/sec04/Lec12FlatMap.java
package de.steklopod.sec04;

import de.steklopod.courseutil.Util;
import de.steklopod.sec04.helper.OrderService;
import de.steklopod.sec04.helper.UserService;

import java.io.BufferedReader;

public class Lec12FlatMap {

    public static void main(String[] args) {

        BufferedReader reader;


        UserService.getUsers()
                .flatMap(user -> OrderService.getOrders(user.getUserId())) // mono / flux
               // .filter(p -> p > 10)
                .subscribe(Util.subscriber());


        Util.sleepSeconds(60);


    }


}

File: src/main/kotlin/de/steklopod/sec04/Lec12FlatMap.kt
package de.steklopod.sec04

import de.steklopod.sec04.helper.OrderService
import de.steklopod.sec04.helper.User
import de.steklopod.sec04.helper.UserService
import de.steklopod.utils.Util.sleepSeconds
import de.steklopod.utils.Util.subscriber
import java.io.BufferedReader

object Lec12FlatMap {
    @JvmStatic
    fun main(args: Array<String>) {
//        var reader: BufferedReader

        UserService.getUsers()
            .flatMap { user: User -> OrderService.getOrders(user.userId) }
            // .filter(p -> p > 10)
            .subscribe(subscriber())
        sleepSeconds(60)
    }
}
